### ISSUE TITLE
Failed to accept incoming connection error when sending many requests

### ISSUE THREAD

**Issue by izgeri** [cyberark/secretless-broker#573]
Steps to reproduce:
- build secretless locally: `./bin/build_darwin`
- run postgres backend (`-d` ensures it runs in the background):
  ```
  pushd test/pg_handler
    docker build -t sample-pg -f Dockerfile.pg .
    docker run -d -p 5432:5432 sample-pg
  popd
  ```

- check if pg is running: `psql -h localhost -p 5432 -U test dbname=postgres`
  you can run `select count(*) from test.test;` and then `\q` to exit (edited) 
- create a secretless.yml file in the project root that has:
  ```
  listeners:
    - name: pg_tcp
      protocol: pg
      address: 0.0.0.0:15432

  handlers:
     - name: pg_via_tcp
      listener: pg_tcp
      debug: true
      credentials:
        - name: address
           provider: env
           id: PG_ADDRESS
        - name: username
          provider: literal
          id: test
        - name: password
          provider: env
          id: PG_PASSWORD
  ```

- run secretless by calling
  `PG_ADDRESS=localhost:5432/postgres PG_PASSWORD=test ./dist/darwin/amd64/secretless-broker -f secretless.yml` 

- create a script that contains
  ```
  #!/bin/bash

  set -euo pipefail

   for i in `seq 1 1000`; do
     command=$(psql -h localhost -p 15432 dbname=postgres -c "select count(*) from test.test")
     echo "$i"
   done
   ```

   and run it

you will see your connections start to fail, and the secretless logs will show:

```
2018/12/19 14:21:11 Connecting client 127.0.0.1:62261 to backend 127.0.0.1:5432
2018/12/19 14:21:11 source 127.0.0.1:62261 closed for destination 127.0.0.1:5432
2018/12/19 14:21:11 Handling connection &{{0xc0005d2c00}}
2018/12/19 14:21:11 h.Client version : 80877103, (SSL mode: true)
2018/12/19 14:21:11 Handling connection &{{0xc0005d2c80}}
2018/12/19 14:21:11 h.Client version : 196608, (SSL mode: false)
2018/12/19 14:21:11 PG backend connection parameters: map[password:test address:localhost:5432/postgres username:test]
2018/12/19 14:21:17 WARN: Failed to accept incoming pg connection: %!(EXTRA *net.OpError=accept tcp [::]:15432: accept: too many open files)
```

when I killed the secretless process I had ~250 `Handler shutting down...` lines show up

**Comment by izgeri** [comment]
This has been resolved in the `pg/to-verify-ca` branch that will be merged in 0.6.0 - leaving open until the final PR has been reviewed to verify.

To reproduce in that branch:
Steps to reproduce:
- build secretless locally: `./bin/build_darwin`
- run postgres backend with SSL enabled:
  ```
  cd test/pg_handler
  docker build -t sample-pg -f Dockerfile.pg .
  docker run -d -p 5432:5432 sample-pg  -c ssl=on -c ssl_cert_file=/var/lib/postgresql/server.crt -c ssl_key_file=/var/lib/postgresql/server.key
  ```

- check if pg is running: `psql -h localhost -p 5432 dbname=postgres -U test -c "select count(*) from test.test"`
- create a secretless.yml file in the project root that has:
  ```
  listeners:
    - name: pg_tcp
      protocol: pg
      address: 0.0.0.0:15432

  handlers:
     - name: pg_via_tcp
      listener: pg_tcp
      debug: true
      credentials:
        - name: address
           provider: env
           id: PG_ADDRESS
        - name: username
          provider: literal
          id: test
        - name: password
          provider: env
          id: PG_PASSWORD
  ```

- run secretless by calling
  `PG_ADDRESS=localhost:5432/postgres PG_PASSWORD=test ./dist/darwin/amd64/secretless-broker -f secretless.yml` 

- create a script that contains
  ```
  #!/bin/bash

  set -euo pipefail

   for i in `seq 1 1000`; do
     command=$(psql sslmode=disable -h localhost -p 15432 postgres -c "select count(*) from test.test")
     echo "$i"
   done
   ```

   and run it


The secretless logs show that connections are closed after queries complete.

**Comment by izgeri** [comment]
This is resolved, so I'm closing this now. I validated it on the latest master branch this evening following the steps outlined in the description.
